src/main/java/uk/gov/hmcts/reform/et/syaapi/service/utils/ManageCaseRoleServiceUtil.java

```diff
@@ -203,11 +218,171 @@ public static boolean isCaseRoleAssignmentExceptionForSameUser(Exception excepti
 
     public static CaseDetails checkCaseDetailsList(List<CaseDetails> caseDetailsList) {
         if (CollectionUtils.isNotEmpty(caseDetailsList)) {
-            CaseDetails caseDetails = caseDetailsList.get(FIRST_INDEX);
+            CaseDetails caseDetails = caseDetailsList.getFirst();
             if (CASE_STATE_ACCEPTED.equals(caseDetails.getState())) {
                 return caseDetails;
             }
         }
         return null;
     }
+
+    /**
+     * Builds a {@link CaseAssignmentUserRolesRequest} using the provided {@link UserInfo},
+     * {@link CaseDetails}, and case role.
+     *
+     *  <p>
+     *      This method constructs a single {@link CaseAssignmentUserRole} that associates the given user
+     *      with the specified case and role. The constructed role assignment is wrapped in a
+     *      {@link CaseAssignmentUserRolesRequest} and returned.
+     *  </p>
+     *
+     * @param userIdamId  the user IDAM ID
+     * @param caseDetails the case details, including the case ID
+     * @param caseRole    the role to assign to the user for the given case (e.g., "[CLAIMANT]", "[DEFENDANT]")
+     * @return a {@link CaseAssignmentUserRolesRequest} containing the role assignment
+     */
+    public static CaseAssignmentUserRolesRequest buildCaseUserRoleRequestByUserIdamIdCaseDetailsAndCaseRole(
```

Comment:
Just a thought: the method name "buildCaseUserRoleRequestByUserIdamIdCaseDetailsAndCaseRole" is too long. readability being the concern. How about just call it "createCaseUserRoleRequest" and not add params detail in the method name? The params are named in self-explanatory way I think.